server/include/ServerNetwork.h

```diff
@@ -42,7 +42,7 @@ class ServerNetwork
     int iResult;
 
     // table to keep track of each client's socket
-    std::map<unsigned int, SOCKET> sessions;
+    std::vector<SOCKET> sessions;
```

Comment:
when a connection closes, we need the session id to remove the right client's socket. so do not apply premature optimizations here

Reply:
reverted to map